src/clarity-angular/data/datagrid/datagrid-row.spec.ts

```diff
@@ -151,6 +151,82 @@ export default function(): void {
+                const cell = context.clarityElement.querySelector("clr-dg-cell");
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.currentSingle).toBeUndefined();
+
+                // Enabling the rowSelectionMode
+                selectionProvider.rowSelectionMode = true;
+                context.detectChanges();
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.currentSingle).toEqual(context.testComponent.item);
+            });
+
+            it("selects the model on click only when `rowSelectionMode` is enabled (Multi selection)", function() {
+                selectionProvider.selectionType = SelectionType.Multi;
+                context.testComponent.item = {id: 1};
+                context.detectChanges();
+                const cell = context.clarityElement.querySelector("clr-dg-cell");
+                expect(selectionProvider.current).toEqual([]);
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.current).toEqual([]);
+
+                // Enabling the rowSelectionMode
+                selectionProvider.rowSelectionMode = true;
+                context.detectChanges();
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.current).toEqual([context.testComponent.item]);
+
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.current).toEqual([]);
+            });
+
+            it("doesn't select the model on click in Row selection mode in case of menu interaction", function() {
+                selectionProvider.selectionType = SelectionType.Single;
+                selectionProvider.rowSelectionMode = true;
+                context.detectChanges();
+
+                const secondCell = context.clarityElement.querySelectorAll("clr-dg-cell")[1];
+                secondCell.click();
+                context.detectChanges();
+                expect(selectionProvider.currentSingle).not.toBeDefined();
+            });
+
+            it("doesn't have input for selection when hideSelectionColumn is enabled (Single selection)", function() {
+                selectionProvider.selectionType = SelectionType.Single;
+                context.detectChanges();
+
+                let input = context.clarityElement.querySelector("input");
+                expect(input).toBeDefined();
+
+                selectionProvider.hideSelectionColumn = true;
+                context.detectChanges();
+                input = context.clarityElement.querySelector("input");
+                expect(input).toBeNull();
+            });
+
+            it("doesn't have input for selection when hideSelectionColumn is enabled (Multi selection)", function() {
+                selectionProvider.selectionType = SelectionType.Single;
```

Comment:
The description of the test says multi-selection, but you're still using single. Happens to all of us on copy-paste. 😄

Reply:
Fixed. Thanks :)